clang/test/CodeGenSYCL/no_opaque_intelfpga-annotated-field.cpp

```diff
@@ -0,0 +1,95 @@
+// RUN: %clangxx %s -fsycl-device-only -fintelfpga -S -o %t.ll
+// RUN: FileCheck %s --input-file %t.ll
+
+// This test checks that clang emits @llvm.ptr.annotation intrinsic correctly
+// when attribute annotate is applied to a kernel functor field.
+// It also ensures the kernel lambda is not inlined for FPGA.
+
+#include "Inputs/sycl.hpp"
+
+// CHECK: @[[STR1:.*]] = private unnamed_addr addrspace(1) constant [9 x i8] c"my_ann_1\00", section "llvm.metadata"
+// CHECK: @[[STR2:.*]] = private unnamed_addr addrspace(1) constant [9 x i8] c"my_ann_2\00", section "llvm.metadata"
+// CHECK: @[[STR3:.*]] = private unnamed_addr addrspace(1) constant [9 x i8] c"my_ann_3\00", section "llvm.metadata"
+
+
+// CHECK: define dso_local spir_kernel void @_ZTS4fooA
+// CHECK: call void @__itt_offload_wi_start_wrapper()
+// CHECK-NEXT: %_arg_p.addr = alloca i32 addrspace(1)*, align 8
+// CHECK-NEXT: %fooA = alloca %struct.fooA, align 8
+// CHECK-NEXT: %fooA.ascast = addrspacecast %struct.fooA* %fooA to %struct.fooA addrspace(4)*
+// CHECK-NEXT: store i32 addrspace(1)* %_arg_p, i32 addrspace(1)** %_arg_p.addr, align 8, !tbaa !47
+// CHECK: %p = getelementptr inbounds %struct.fooA, %struct.fooA addrspace(4)* %fooA.ascast, i32 0, i32 0
+// CHECK-NEXT: %1 = bitcast i32 addrspace(4)* addrspace(4)* %p to i8 addrspace(4)*
+// CHECK-NEXT: %2 = call i8 addrspace(4)* @llvm.ptr.annotation.p4i8.p1i8(i8 addrspace(4)* %1, i8 addrspace(1)* getelementptr inbounds ([9 x i8], [9 x i8] addrspace(1)* @[[STR1]], i32 0, i32 0)
+// CHECK-NEXT: %3 = bitcast i8 addrspace(4)* %2 to i32 addrspace(4)* addrspace(4)*
+// CHECK-NEXT: %4 = load i32 addrspace(1)*, i32 addrspace(1)** %_arg_p.addr, align 8
+// CHECK-NEXT: %5 = addrspacecast i32 addrspace(1)* %4 to i32 addrspace(4)*
+// CHECK-NEXT: store i32 addrspace(4)* %5, i32 addrspace(4)* addrspace(4)* %3, align 8
+// CHECK-NEXT: call spir_func void @_ZNK4fooAclEv(%struct.fooA addrspace(4)* noundef align 8 dereferenceable_or_null(8) %fooA.ascast)
+// CHECK: call void @__itt_offload_wi_finish_wrapper()
```

Comment:
Would it be possible to simplify this checks to the four or five IR statements that the description has?   I feel like that will keep the test more stable.